src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -497,31 +512,39 @@ def _validate_vm_create_nsg(namespace):
         if check_existence(namespace.nsg, namespace.resource_group_name,
                            'Microsoft.Network', 'networkSecurityGroups'):
             namespace.nsg_type = 'existing'
+            logger.debug("using specified NSG '%s'", namespace.nsg)
         else:
             namespace.nsg_type = 'new'
+            logger.debug("specified NSG '%s' not found. One will be created.", namespace.nsg)
     elif namespace.nsg == '':
         namespace.nsg_type = None
+        logger.debug('no NSG will be used')
     elif namespace.nsg is None:
         namespace.nsg_type = 'new'
+        logger.debug('new NSG will be created')
 
 
 def _validate_vm_create_public_ip(namespace):
     if namespace.public_ip_address:
         if check_existence(namespace.public_ip_address, namespace.resource_group_name,
                            'Microsoft.Network', 'publicIPAddresses'):
             namespace.public_ip_type = 'existing'
+            logger.debug("using existing specified public IP '%s'", namespace.public_ip_address)
         else:
             namespace.public_ip_type = 'new'
+            logger.debug("specified public IP '%s' not found. It will be created.", namespace.public_ip_address)
```

Comment:
"One will be created" is for when you didn't specify anything. "It will be created" is for when you actually specified a name.  So there's a subtle difference.